ICSharpCode.ILSpyX/PdbProvider/PortableDebugInfoProvider.cs

```diff
@@ -161,5 +161,75 @@ public bool TryGetName(MethodDefinitionHandle method, int index, [NotNullWhen(tr
 			}
 			return false;
 		}
+
+		public bool TryGetExtraTypeInfo(MethodDefinitionHandle method, int index, out PdbExtraTypeInfo extraTypeInfo)
+		{
+			var metadata = GetMetadataReader();
+			extraTypeInfo = default;
+
+			if (metadata == null)
+				return false;
+
+			LocalVariableHandle localVariableHandle = default;
+			foreach (var h in metadata.GetLocalScopes(method))
+			{
+				var scope = metadata.GetLocalScope(h);
+				foreach (var v in scope.GetLocalVariables())
+				{
+					var var = metadata.GetLocalVariable(v);
+					if (var.Index == index)
+					{
+						localVariableHandle = v;
+						break;
+					}
+				}
+
+				if (!localVariableHandle.IsNil)
+					break;
+			}
+
+			foreach (var h in metadata.CustomDebugInformation)
+			{
+				var cdi = metadata.GetCustomDebugInformation(h);
+				if (cdi.Parent.IsNil || cdi.Parent.Kind != HandleKind.LocalVariable)
+					continue;
+				if (localVariableHandle != (LocalVariableHandle)cdi.Parent)
+					continue;
+				if (cdi.Value.IsNil || cdi.Kind.IsNil)
+					continue;
+				var kind = metadata.GetGuid(cdi.Kind);
+				if (kind == KnownGuids.TupleElementNames && extraTypeInfo.TupleElementNames is null)
+				{
+					var reader = metadata.GetBlobReader(cdi.Value);
+					var list = new List<string?>();
+					while (reader.RemainingBytes > 0)
+					{
+						int length = reader.IndexOf(0);
+						string s = reader.ReadUTF8(length);
+						reader.ReadByte();
```

Comment:
Should probably use `ReadUTF8StringNullTerminated` instead or add a comment why the extra `ReadByte` is necessary.

Reply:
`ReadUTF8StringNullTerminated` is not available in `ILSpyX` project as it's defined in `ILSpy`. I will go ahead and add a comment stating that this extra `ReadByte` is responsible for reading the null terminator.

Reply:
Ah, sorry, that method only exists in ILSpy.